File: iOS/TUILiveKit/VoiceRoomDefine.swift
//
//  VoiceRoomDefine.swift
//  TUILiveKit
//
//  Created by jack on 2024/10/8.
//

import Foundation
import RTCRoomEngine

@objcMembers
public class VoiceRoomDefine: NSObject {
    
    public static let MAX_CONNECTED_VIEWERS_COUNT: Int = 10
    
}

@objcMembers
public class CreateRoomParams: NSObject {
    public var roomName: String
    public var maxAnchorCount: Int
    public var seatMode: TUISeatMode
    
    public override init() {
        self.roomName = ""
        self.maxAnchorCount = VoiceRoomDefine.MAX_CONNECTED_VIEWERS_COUNT
        self.seatMode = .applyToTake
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/voiceroom/VoiceRoomDefine.java
package com.trtc.uikit.livekit.voiceroom;

import com.tencent.cloud.tuikit.engine.room.TUIRoomDefine;

import java.io.Serializable;

public class VoiceRoomDefine {
    public static final int MAX_CONNECTED_VIEWERS_COUNT = 10;

    public static class CreateRoomParams implements Serializable {
        public String                 roomName       = "";
        public int                    maxAnchorCount = MAX_CONNECTED_VIEWERS_COUNT;
        public TUIRoomDefine.SeatMode seatMode       = TUIRoomDefine.SeatMode.APPLY_TO_TAKE;
    }
}
